### PR TITLE
Metrics checks in regular REST API calls

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Metrics checks in regular REST API calls
Not all possible cases are tested, it's just beginning, some utility functions etc.

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] end to end tests

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: tests/e2e/test_api.py [hunk 1]
```diff
@@ -68,11 +163,18 @@ def test_invalid_question():
     }
     assert response.json() == expected_json
 
+    # new_llm_calls = get_model_provider_counter_value(client, "llm_calls_total",
+    #                                                 "ibm/granite-13b-chat-v2",
+    #                                                 "bam")
+    # check_counter_increases(endpoint, old_llm_calls, new_llm_calls)
```
Comment (onmete) [inline tests/e2e/test_api.py:169]: What is the deal with that commented code?

File: tests/e2e/test_api.py [hunk 2]
```diff
@@ -81,11 +183,19 @@ def test_query_call_without_payload():
     # so let's do just primitive check
     assert "missing" in response.text
 
+    counter = get_rest_api_counter_value(client, "rest_api_calls_total", endpoint, 422)
```
Comment (onmete) [inline tests/e2e/test_api.py:186]: At first sight, I thought the `422` is the actual counter value, not the status code. I would do `status_code=422`, but up to you. Not a big deal.

Comment (tisnik) [reply tests/e2e/test_api.py:186]: yup that needs change, indeed

File: tests/e2e/test_api.py (hunk 1 above)
Comment (tisnik) [reply tests/e2e/test_api.py]: yup, will be added later. now I removed them

File: tests/e2e/test_api.py (hunk 2 above)
Comment (tisnik) [reply tests/e2e/test_api.py]: fixed, now it should be more readable

**Pull request merged**